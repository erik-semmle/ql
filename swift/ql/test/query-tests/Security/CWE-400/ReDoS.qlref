queries/Security/CWE-400/ReDoS.ql
