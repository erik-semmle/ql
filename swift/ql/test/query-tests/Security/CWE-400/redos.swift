//codeql-extractor-options: -module-name ReDoS

import Foundation

let bad1 = try! NSRegularExpression(pattern: "(a|aa)*Y", options: [])