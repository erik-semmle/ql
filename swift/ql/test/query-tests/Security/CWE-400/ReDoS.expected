| redos.swift:5:47:5:53 | (a\|aa)* | This part of the regular expression may cause exponential backtracking on strings containing many repetitions of 'aa'. |
